connections.md

```diff
@@ -2,254 +2,7 @@
-* The `did` indicates the DID being exchanged.
-
-###### Correlating requests to invitations
-
-An invitation is presented in one of two forms:
-
-* An explicit OutOfBand invitation with its own `@id`.
-* An [implicit](#implicit-invitation) invitation contained in a DID document's [`service`](https://w3c-ccg.github.io/did-spec/#service-endpoints) attribute.
-
-When a `request` responds to an explicit invitation, its `~thread.pthid` MUST be equal to the `@id` property of the invitation.
-
-When a `request` responds to an implicit invitation, its `~thread.pthid` MUST contain a [DID URL](https://w3c-ccg.github.io/did-spec/#dfn-did-url) that resolves to the specific `service` on a DID document that contains the invitation.
-
-TODO: Include examples using a ledger resolvable DID.
-
-**Example referencing an explicit invitation**
-
-```json
-{
-  "@id": "a46cdd0f-a2ca-4d12-afbf-2e78a6f1f3ef",
-  "@type": "https://didcomm.org/didexchange/2.0/request",
-  "~thread": { "pthid": "032fbd19-f6fd-48c5-9197-ba9a47040470" },
-  "label": "Bob",
-  "did": "<did invitee is sending to inviter>"
-}
-```
-
-**Example referencing an implicit invitation**
-
-```json
-{
-  "@id": "a46cdd0f-a2ca-4d12-afbf-2e78a6f1f3ef",
-  "@type": "https://didcomm.org/didexchange/2.0/request",
-  "~thread": { "pthid": "did:example:21tDAKCERh95uGgKbJNHYp#invitation" },
-  "label": "Bob",
-  "did": "<did invitee is sending to inviter>"
-}
-```
-
-
-###### Request Transmission
-
-The Request message is encoded according to the standards of the Encryption Envelope, using the `recipientKeys` present in the invitation.
-
-If the `routingKeys` attribute was present and non-empty in the invitation, each key must be used to wrap the message in a forward request, then encoded in an Encryption Envelope. This processing is in order of the keys in the list, with the last key in the list being the one for which the `serviceEndpoint` possesses the private key.
-
-The message is then transmitted to the `serviceEndpoint`.
-
-We are now in the `requested` state.
-
-###### Request processing
-
-After receiving the exchange request, the _inviter_ evaluates the provided DID and DID Doc according to the DID Method Spec.
-
-The _inviter_ should check the information presented with the keys used in the wire-level message transmission to ensure they match.
-
-The _inviter_ MAY look up the corresponding invitation identified in the request's `~thread.pthid` to determine whether it should accept this exchange request.
-
-If the _inviter_ wishes to continue the exchange, they will persist the received information in their wallet. They will then either update the provisional service information to rotate the key, or provision a new DID entirely. The choice here will depend on the nature of the DID used in the invitation.
-
-The _inviter_ will then craft an exchange response using the newly updated or provisioned information.
-
-###### Request Errors
```

Comment:
I am curious about the difference in errors originating from OOB/DID Exchange vs the DIDComm v2 model?

Reply:
Can you pick an example to discuss?

Reply:
If I were to generalize the thought: How do we differentiate between a protocol-layer errors vs lower-layer concerns that were previously in these pre-step connection protocols. As an example, an unsupported DID method happened to be used (or unsupported key type or unsupported endpoint, etc).

Of course there is also possibilities such as unsupported protocols (or versions), but these would be cases that could happen regardless.

Reply:
I don't believe OOB has anything in the spec about any of the errors you mentioned. Those are certainly important to think about, but I don't think there is a difference there between OOB/DIDExchange and 0RTT.